### PR TITLE
add first reusable workflows

### PR DESCRIPTION
**Pull request by michaelamattes**
add reusable workflows for terraform linting and create release draft

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+name: release
+
+on:
+  workflow_call:
+    inputs:
+      readme:
```
Comment (schurzi) [inline .github/workflows/release.yml:7]: I think a more specific name would be better here e.g. `readme_type`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        required: false
+        type: string
+      files:
```
Comment (schurzi) [inline .github/workflows/release.yml:10]: also here, maybe `update_files`(?)

**Review by schurzi — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (rndmh3ro) [inline .github/workflows/release.yml:7]: * Can you find a better name for these inputs?
* Can you add a description what these inputs do?
* You can use options to define what can be inserted.
* 
https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/

Comment (schurzi) [reply .github/workflows/release.yml:7]: hmm, the linked blog states, that these options are for manually triggered workflows. We can try to check if this also works for workflow calls.
Your comment made me think again. Maybe we should drop the parameters and instead create separate fixed workflows. We maybe have 2-3 different release workflows. That is an amount where copying stuff still seems ok.

Comment (michaelamattes) [reply .github/workflows/release.yml:7]: Maybe something like this ?

```on:
  workflow_call:
    inputs:
      readme_docs:
        required: false
        type: string
        description: creates a README over a tool based Github action
        options:
        - terraform
      commit_files:
        required: false
        type: string
        description: specific files which should be added during the release build e.g. README.md, see https://github.com/github-actions-x/commit/blob/master/action.yml - files option
```

example for usage can be found here https://github.com/T-Systems-MMS/terraform-template/blob/main/.github/workflows/release.yml

Comment (michaelamattes) [reply .github/workflows/release.yml:7]: I'll take a looku to make seperate release.yml for terraform

File: .github/workflows/terraform_release.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+---
+name: terraform_release
+
+jobs:
```
Comment (schurzi) [inline .github/workflows/terraform_release.yml:4]:
```suggestion
name: terraform_release

on:
  workflow_call:

jobs:
```

**Review by schurzi — CHANGES_REQUESTED** [review]

**Review by schurzi — APPROVED** [review]

**Comment by schurzi** [comment]
Awesome! Thank you.

**Pull request merged**